salt/modules/telemetry.py

```diff
@@ -277,9 +277,9 @@ def create_alarm(deployment_id, metric_name, data, api_key=None, profile="teleme
         _update_cache(deployment_id, metric_name, response.json())
     else:
         log.error(
-            str('Failed to create alarm on metric: %s in deployment %s: payload: %s'),  # future lint: disable=blacklisted-function
-            salt.utils.stringutils.to_str(metric_name),
-            salt.utils.stringutils.to_str(deployment_id),
+            six.text_type('Failed to create alarm on metric: %s in deployment %s: payload: %s'),  # future lint: disable=blacklisted-function
+            salt.utils.stringutils.to_unicode(metric_name),
+            salt.utils.stringutils.to_unicode(deployment_id),
```

Comment:
This log message should be changed. This was originally done intentionally since we're including the results of a json.dumps (which is a `str` type on both Python 2 and Python 3). This was done before our log handler was modified to gracefully handle str/unicode mismatches, so this kind of stuff is no longer necessary. This log message can be rewritten as:

```python
log.error(
    'Failed to create alarm on metric: %s in deployment %s: payload: %s'
    metric_name, deployment_id, salt.utils.json.dumps(post_body)
)
```

You may also need to remove the `salt.utils.stringutils` import as part of this change.